src/align.rs

```diff
@@ -87,3 +90,79 @@ def_align!(8, Align8);
 def_align!(16, Align16);
 def_align!(32, Align32);
 def_align!(64, Align64);
+
+/// A `Vec` that uses a 64 byte aligned allocation.
+///
+/// Only works with `Copy` types so that we don't have to handle drop logic.
+#[derive(Default)]
+pub struct AlignedVec64<T: Copy> {
+    inner: Vec<MaybeUninit<Align64<[u8; 64]>>>,
+    len: usize,
+    _phantom: std::marker::PhantomData<T>,
+}
+
+impl<T: Copy> AlignedVec64<T> {
+    /// Returns the number of elements in the vector.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn as_ptr(&self) -> *const T {
+        self.inner.as_ptr().cast()
+    }
+
+    pub fn as_mut_ptr(&mut self) -> *mut T {
+        self.inner.as_mut_ptr().cast()
+    }
+
+    /// Extracts a slice containing the entire vector.
+    pub fn as_slice(&self) -> &[T] {
+        // SAFETY: The first `len` elements of the buffer have been initialized.
+        unsafe { slice::from_raw_parts(self.as_ptr(), self.len) }
+    }
+
+    /// Extracts a mutable slice of the entire vector.
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        // SAFETY: The first `len` elements of the buffer have been initialized.
+        unsafe { slice::from_raw_parts_mut(self.as_mut_ptr(), self.len) }
+    }
+
+    pub fn resize_with<F>(&mut self, new_len: usize, mut f: F)
+    where
+        F: FnMut() -> T,
+    {
+        let len = self.len();
+
+        if new_len > len {
+            // Resize the underlying vector to have enough capacity for the new length.
+            let new_chunks = (mem::size_of::<T>() * new_len / 64) + 1;
+            self.inner.resize_with(new_chunks, MaybeUninit::uninit);
+
+            // Initialize the new elements past `len`.
+            for offset in len..new_len {
+                // SAFETY: We've allocated enough space to write up to `new_len` elements into the buffer.
+                unsafe {
+                    self.as_mut_ptr().add(offset).write(f());
+                }
+
+                self.len += 1;
+            }
+        } else {
+            self.len = new_len;
```

Comment:
I moved the update to `self.len` to the end of the function. The original logic for updating by 1 each loop came from the original `Vec` logic in order to be panic safe when running destructors, but we don't need support for that and so can do the simpler thing.

Likewise I've updated the function to always call `resize_with` on the underlying vector to make sure it has a chance to reduce its size.

Reply:
Ah, that was the panic safety stuff you were talking about.  I didn't realize it came from `Vec`'s sources?  Is it still an issue if `f` can panic, though, even though the destructor can't?  We can similarly avoid that issue by changing `fn resize_with` to `fn resize` with a `value: T` arg that is copied, right?

Reply:
I don't think panic safety is an issue here because `T` doesn't need to be dropped, and I think the issues around panic safety all relate to making sure invalid data isn't dropped (`Vec` generally leaks destructors in the case of a panic). But you're right that `resize` makes more sense than `resize_with` since `T: Copy`. I'll make that change.